### PR TITLE
Add rule for practice exercise `zebra-puzzle`

### LINKED ISSUE #54: Write analysis for practice exercise `zebra-puzzle`
It's incredibly easy to cheat in this exercise, so we should make sure the functions are not exactly like`drinksWater = Just Norwegian`.

### PR DESCRIPTION
**Pull request by jiegillet**
Closes #54 

[Sister PR over here](https://github.com/exercism/website-copy/pull/2283).
There are countless workarounds for bypassing this rule, but I think there is no meaning in pursuing them.

### CODE REVIEW

File: src/Exercise/ZebraPuzzle.elm
Comment (ceddlyburge) [inline src/Exercise/ZebraPuzzle.elm:39]: This checks that the expression for the function is 'Just' something, right?

**Review by ceddlyburge — APPROVED** [review]
I can't reason all this through, but it looks logical, and the tests show that works as intended.

File: src/Exercise/ZebraPuzzle.elm
Comment (jiegillet) [reply src/Exercise/ZebraPuzzle.elm:39]: Exactly. If the analyzer runs, we know that the tests pass, so there is no point of checking the actual content.

**Comment by jiegillet** [comment]
@exercism/maintainers-admin Can I get an approval? 💛

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**
